src/System.Private.DataContractSerialization/src/System/Xml/XmlUTF8TextWriter.cs

```diff
@@ -101,7 +101,7 @@ public XmlUTF8NodeWriter(bool[] isEscapedAttributeChar, bool[] isEscapedElementC
         new public void SetOutput(Stream stream, bool ownsStream, Encoding encoding)
         {
             Encoding utf8Encoding = null;
-            if (encoding != null && encoding == Encoding.UTF8)
+            if (encoding != null && encoding.CodePage == Encoding.UTF8.CodePage)
```

Comment:
Can we use encoding.Equals(Encoding.UTF8) for equality checking here? I think that includes the .CodePage equality check as well.